File: 0392-Is Subsequence/0392-Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int m = s.size();
        int n = t.size();
        int p1 = 0, p2 = 0;
        while (p1 < m && p2 < n)
        {
            if (s[p1] == t[p2])
            {
                ++p1;
            }
            ++p2;
        }
        
        return p1 == m;
    }
};

/* Follow up: Preprocess string t, save each letter with index array,
find target letter in index array with binary search.*/

File: 0301-0400/0392-Is Subsequence/0392-Is Subsequence.py
class Solution:
    def isSubsequence(self, s: 'str', t: 'str') -> 'bool':
        p1 = p2 = 0
        while p1 < len(s) and p2 < len(t):
            if s[p1] == t[p2]:
                p1 += 1
            p2 += 1
        return p1 == len(s)
